### DISCUSSION TITLE
[docs] Continue to flesh out gatsby-node api functions descriptions

### DISCUSSION THREAD

**Discussion by lannonbr** [gatsbyjs/gatsby#31217] [Ideas / Feature Requests]
## Summary

We have currently documented each of the various Node APIs here: https://www.gatsbyjs.org/docs/node-apis

That said, a large portion of them are just the name and a small description of it. What would be useful would be to add further context about each one that can include things like
* Parameters specific to this API 
* an example of using the API

### Basic example

the `onCreateWebpackConfig` API is a good example to describe what the API is, what params are passed in, and an example of using it. 

https://www.gatsbyjs.org/docs/node-apis/#onCreateWebpackConfig
![Screen Shot 2019-03-19 at 9 47 50 AM](https://user-images.githubusercontent.com/3685876/54610793-1736e200-4a2c-11e9-9f9f-de60c56b1997.png)

**Comment by sampittko** [comment]
I would like to document onCreatePage API, should I create another issue referencing this one but more specific?
